### DISCUSSION TITLE
feature request: support symbolic constants for `base_address`, (do not require it to be a number)

### DISCUSSION THREAD

**Discussion by v0lker** [esynr3z/corsair#108] [Questions]
i think it would be nice to be able to write this in the `config`
```
[globcfg]
base_address = MY_SYMBOLIC_CONSTANT
```

or maybe even
```
[globcfg]
base_address = MY_SYMBOLIC_CONSTANT + SOME_OFFSET
```

and to get this in the C headers:

```
#define REGS_BASE_ADDR MY_SYMBOLIC_CONSTANT
```

or

```
#define REGS_BASE_ADDR (MY_SYMBOLIC_CONSTANT + SOME_OFFSET)
```

respectively. maybe we don't need to be able to handle arbitrary strings, but it's probably the same amount of effort (?)

does anybody else like the idea?

**Comment by v0lker** [comment]
implementation could look like this: https://github.com/esynr3z/corsair/pull/51

**Comment by esynr3z** [comment]
It's a nice feature, really. But it opens a portal to symbolic calculations instead of concrete numbers, which is not an easy thing to implement and maintain within all templates.

I've also answered to the quite similar topic [here](https://github.com/esynr3z/corsair/issues/49#issuecomment-2408075945), if you are interested.

I thinks this won't be implemented soon, but, indeed, this has to be thoroughly thought.

**Reply by v0lker** [reply]
> it opens a portal to symbolic calculations instead of concrete numbers

if we're talking about implementing actual symbolic computation, as #49 seems to suggest, then, yes, that would look at least a little bit more complicated. in my use case, i only needed the base address to be a string, see the PR #51 . with that small change i could build everything by passing in the actual base address at compile time.

if you solve the general case, then this approach here (#108 and #51) will be obsolete  --  your call...
